pkg/utils/goroutine_pool.go

```diff
@@ -0,0 +1,57 @@
+package utils
+
+import (
+	"fmt"
+	"sync"
+)
+
+type GoroutinePool struct {
```

Comment:
add some unit test for this GoroutingPool

Reply:
I fixed it